### PR TITLE
Replace Bullet with Prosopite

### PR DESCRIPTION
**Pull request by santib**
Evaluate replacing Bullet with Prosopite.

Remove `Bullet`: Bullet has many false positives, causes a lot of headaches, doesn't work properly with Rails 7 (e.g. in some of our projects it's not detecting existing N+1s), the code is complex, and it's not well maintained.

Add `Prosopite`: It's like Bullet but its code is much simpler. We tested it in some of our projects and it worked well. Additionally, it's [recommended by Nate Berkopec](https://twitter.com/nateberkopec/status/1674055978856087553), who is an expert in these matters.

Add `Rails' strict_loading`: Rails 6.1 added this feature, which is good and useful for detecting N+1s, but it still has some limitations in its functionality that prevent us from using it instead of Prosopite. What we can start doing is using it to log N+1s (in all environments), and later on, if the feature continues to mature, it might be able to fully replace Prosopite.

### CODE REVIEW

**Review by JulianPasquale — APPROVED** [review]

**Comment by TimoPeraza** [comment]
Can you update `readme` changing `[Bullet](https://github.com/flyerhzm/bullet) help to kill N+1`?

**Review by TimoPeraza — APPROVED** [review]

**Review by sebastiancaraballo — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -69,7 +69,6 @@ To illustrate, `bin/rails console` will run the console in the docker container
 - [Annotate](https://github.com/ctran/annotate_models) for doc the schema in the classes
 - [Better Errors](https://github.com/charliesome/better_errors) for a better error page
 - [Brakeman](https://github.com/presidentbeef/brakeman) for static analysis security
-- [Bullet](https://github.com/flyerhzm/bullet) help to kill N+1
```
Comment (santib) [inline README.md:72]: @TimoPeraza you mean this? or did I miss any other place?

**Review by PerezIgnacio — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (TimoPeraza) [reply README.md:72]: Yes, that one

Comment (santib) [reply README.md:72]: ok it's already removed

**Review by juanmferreira93 — APPROVED** [review]

**Review by claramaffei — APPROVED** [review]

**Review by santiagodiaz — APPROVED** [review]

**Pull request merged**